src/modules/mod/slash-commands/profile-presets.command.ts

```diff
@@ -0,0 +1,307 @@
+          opt.setName("id").setDescription("id of the preset").setRequired(true)
+        )
+    )
+    .addSubcommand((sub) =>
+      sub
+        .setName(ProfileCommandSubCommandsTypes.LIST)
+        .setDescription("list all presets")
+    )
+    .addSubcommand((sub) =>
+      sub
+        .setName(ProfileCommandSubCommandsTypes.INFO)
+        .setDescription("Overall info on JSON etc.")
+    ),
+})
+export class ProfilePresetsCommand extends BaseCommand<CmdType.SLASH_COMMAND> {
+  async execute(arg: CmdArg<CmdType.SLASH_COMMAND>) {
+    try {
+      const subCommandName = arg.options.getSubcommand(true);
+      const { CREATE, UPDATE, DELETE, LIST, INFO } =
+        ProfileCommandSubCommandsTypes;
+
+      const commands: {
+        [key: string]: (arg: CmdArg<CmdType.SLASH_COMMAND>) => Promise<unknown>;
+      } = {
+        [CREATE]: this.createPreset.bind(this),
+        [UPDATE]: this.updatePreset.bind(this),
+        [DELETE]: this.deletePreset.bind(this),
+        [LIST]: this.listPresets.bind(this),
+        [INFO]: this.infoPreset.bind(this),
+      };
+
+      await arg.deferReply({ ephemeral: true });
+
+      if (commands[subCommandName]) return commands[subCommandName](arg);
+    } catch (err) {
+      logger.error(err);
+    }
+  }
+
+  // ! We will reuse this method in another command when we will implement ability for generic users to buy custom profile presets.
+  public async createPreset(
+    arg: CmdArg<CmdType.SLASH_COMMAND>,
+    isCustom = false
+  ) {
+    const json = arg.options.getString("json", true);
+    if (!isValidJson(json))
+      return await this.sendError("JSON you provided is not valid", arg);
+
+    const parsedJson = JSON.parse(json);
+
+    // TODO: Add more checks for other fields (should create a helper for that)
+    // src: https://discordjs.guide/popular-topics/embeds.html#embed-limits
+    if (!parsedJson.description) {
+      return this.sendError(
+        "JSON you provided doesn't have description field",
+        arg
+      );
+    } else if (parsedJson.description.length > 4096) {
+      return this.sendError("Description is too long. Max length is 4096", arg);
+    } else if (parsedJson.description.length < 1) {
+      return this.sendError("Description is too short", arg);
+    }
+
+    // Normalize JSON for storing in db (remove spaces etc.)
+    const modifiedJson = JSON.stringify(JSON.parse(json));
+
+    const data = {
+      updated_by: arg.user.id,
+      json: modifiedJson,
+    };
+
+    if (isCustom) {
+      await ProfilePresetEntity.save({ user: { uid: arg.user.id }, ...data });
+    } else {
+      await ProfilePresetEntity.save({ ...data });
+    }
+
+    return this.sendSuccess("New profile preset created!", arg);
+  }
+
+  public async updatePreset(arg: CmdArg<CmdType.SLASH_COMMAND>) {
+    const id = arg.options.getString("id", true);
+    const json = arg.options.getString("json", true);
+
+    if (!id.match(/^[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}$/))
```

Comment:
Not really accurate, but as an example: https://i.imgur.com/IbzR7vv.png
I don't think it _that_ slow in comparison to db request. Also, should mention that we should really overthink that +-10ms, because update/remove should be used only by moderators very rare.

Reply:
Good take, agree with you